src/VisualStudio/CSharp/Impl/CodeModel/CSharpCodeModelService.cs

```diff
@@ -3021,6 +3021,14 @@ protected override ITypeSymbol GetTypeSymbolFromPartialName(string partialName,
 
         public override ITypeSymbol GetTypeSymbolFromFullName(string fullName, Compilation compilation)
         {
+            // Some clients pass us an assembly qualified name, which we don't handle.  Just lop off the
+            // assembly info if so.
+            var index = fullName.IndexOf(',');
+            if (index > 0)
+            {
+                fullName = fullName.Substring(0, index);
+            }
+
             ITypeSymbol typeSymbol = compilation.GetTypeByMetadataName(fullName);
 
             if (typeSymbol == null)
```

Comment:
The fallback *is* necessary. For example, DotNetNameFromLanguageSpecific can be passed `System.Collections.Generic.List(Of Integer)`, or `int`, or any number of things that aren't metadata names. Does `Compilation.GetTypeByMetadataName` handle assembly qualification? If so, maybe the right thing to do is remove the first bit of code above that (potentially) drops the assembly qualification.